include/ensmallen_bits/callbacks/early_stop_at_min_loss.hpp

```diff
@@ -12,12 +13,15 @@
 #ifndef ENSMALLEN_CALLBACKS_EARLY_STOP_AT_MIN_LOSS_HPP
 #define ENSMALLEN_CALLBACKS_EARLY_STOP_AT_MIN_LOSS_HPP
 
+#include <functional>
+
 namespace ens {
 
 /**
  * Early stopping to terminate the optimization process early if the loss stops
  * decreasing.
  */
+template<typename MatType = arma::mat>
```

Comment:
Crap, actually, I just realized, to make this change would force a major version bump:

https://github.com/mlpack/ensmallen/blob/master/UPDATING.txt

I think we can fix this by doing the following:

 - Rename this class to `EarlyStopAtMinLossType`.
 - Create a typedef so that `EarlyStopAtMinLoss` is `EarlyStopAtMinLossType<arma::mat>`.
 - Update the documentation (I'll provide a suggestion).
 - Add a comment here suggesting that we should change `EarlyStopAtMinLossType` to `EarlyStopAtMinLoss` for ensmallen 3.10.0.